Storage/SQL/SQLiteHistoryRecommendations.swift

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+import XCGLogger
+import Deferred
+
+extension SQLiteHistory: HistoryRecommendations {
+    public func getHighlights() -> Deferred<Maybe<Cursor<Site>>> {
+        let limit = 20
+        let microsecondsPerMinute: Timestamp = 60_000_000 // 1000 * 1000 * 60
+        let now = NSDate.nowMicroseconds()
+        let thirtyMinutesAgo = now - 30 * microsecondsPerMinute
+        let threeDaysAgo = now - (60 * microsecondsPerMinute) * 24 * 3
+        let bookmarkLimit = 1
+        let historyLimit = limit - bookmarkLimit
+
+        let highlightProjection = "historyID, url, title, guid, visitCount, visitDate, iconID, iconURL, iconType, iconDate, iconWidth"
+
+        let nonRecentHistory = [
+            "SELECT * FROM (",
+            "   SELECT \(TableHistory).id as historyID, url, title, guid, visitDate, (SELECT COUNT(1) FROM \(TableVisits) WHERE s = \(TableVisits).siteID) AS visitCount",
+            "   FROM (",
+            "       SELECT siteID AS s, max(date) AS visitDate",
+            "       FROM \(TableVisits)",
+            "       WHERE date < \(thirtyMinutesAgo)",
```

Comment:
`thirtyMinutesAgo` is the only dynamic part of this query. If you extract that as a value in `args`, the compiler should be able to avoid dynamically concatenating this string. (Same below for `threeDaysAgo`.)

In the future, if we need to squeeze more perf out of the system, that will also allow us to pre-parse statements rather than parsing SQL each time we fetch.